backend/djangoindia/bg_tasks/send_update.py

```diff
@@ -0,0 +1,24 @@
+from celery import shared_task
+from django.core.mail import send_mail
+from djangoindia.db.models.update import Update
+from django.conf import settings
+
+@shared_task
+def send_update_email(update_id):
+    update = Update.objects.get(pk=update_id)
+    for subscription in update.recipients.all():
+        try:
+            send_mail(
+                subject=f"DjangoIndia Update: {update.name}",
+                message=update.content,
+                from_email=settings.DEFAULT_FROM_EMAIL,
+                recipient_list=[subscription.email],
+                fail_silently=False,
+            )
+            update.sent_successfully.add(subscription)
+        except Exception as e:
+            #stored failted email
+            update.failed_to_send.add(subscription)
+            print(f"Failed to send email to {subscription.email}: {str(e)}")
+    
+    update.save()
```

Comment:
the db operations should ideally happen in a separate storage classes

Reply:
Yes, ideally we should create a separate utility class to handle these operations. I would suggest writing a class, and then separate methods (static methods) to log success and failed cases.